common/scripts/lib/_helpers.sh

```diff
@@ -313,20 +319,36 @@ _validate_ip() {
 
 __set_admiral_ip() {
   __process_msg "Setting value of admiral IP address"
-  local admiral_ip='127.0.0.1'
 
-  __process_success "Please enter your current IP address. This will be the address at which you access the installer webpage. Type D to set default (127.0.0.1) value."
+  __process_msg "List of your machine IP addresses:"
+
+  local machine_ips=('127.0.0.1')
+
+  if [[ "$MACHINE_IPS" ]]; then
+    machine_ips+=( $MACHINE_IPS )
+  fi
+
+  for (( i = 1 ; i < ${#machine_ips[@]} + 1 ; i++ )) do
+    __process_msg "$i - (${machine_ips[$i-1]})"
+  done
+
+  __process_success "Please pick one of the IP addresses from the above list by selecting the index (1, 2, etc) or enter a custom IP"
   read response
 
-  if [ "$response" != "D" ] && [ "$response" != "d" ]; then
+  if [[ $response =~ ^[0-9]+$ ]]; then
+    if [ ${machine_ips[$response-1]} ]; then
+      export ADMIRAL_IP=${machine_ips[$response-1]}
```

Comment:
add logger afeter this, `admiral ip is set to ...`

Reply:
we want more verbosity here to make sure user knows exactly what is happening with each step they're taking. we can later hide some of the redundant stuff but the basic assumption should be that user should be shown everything

Reply:
ok, i will update it.